src/styles/getMuiTheme.js

```diff
@@ -4,6 +4,21 @@ import ColorManipulator from '../utils/color-manipulator';
 import autoPrefix from './auto-prefix';
 import lightBaseTheme from './baseThemes/lightBaseTheme';
 import zIndex from './zIndex';
+import {autoprefixer, rtl} from './transformers';
+import compose from 'lodash.flowright';
+import warning from 'warning';
+
+const CALLED_ONCE = Symbol('prepareStyles/calledOnce');
+
+const callOnce = (transformer) => (style) => {
```

Comment:
Mm, I hear you! You're right.

My initial thought was that the performance cost of just returning the object you passed in is *less* than re-running the whole chain again on the same object and would make this approach more performant in production. But that is only true when compare when the function is used twice. If it is just ran once, the check is unnecessary and would reduce performance.

Also, another point I suppose is this function really is primarily for us not for users of the library. So as long as we're catching the warning then in production it should be fine. It'd be interesting if we could just write a test for this and be done with it.